### PR TITLE
[Java][jaxrs] Add getter to ApiException templates

### LINKED ISSUE #6830: [BUG][JAVA] ApiException error code is private without getter
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] What's the version of OpenAPI Generator used?
- [x] Have you search for related issues/PRs?
- [x] What's the actual output vs expected output?
- [x] [Optional] Bounty to sponsor the fix ([example](https://www.bountysource.com/issues/66123212-javascript-client-produces-a-wrong-object-for-a-string-enum-type-that-is-used-with-ref))

##### Description

The [ApiException.mustache template](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/JavaJaxRS/resteasy/ApiException.mustache) provides a useful exception with code and name, but unfortunately the code is private (well, that's good actually) but there is no getter to access it, so we cannot really use it...

##### openapi-generator version

4.3.1

##### OpenAPI declaration file content or url

https://github.com/OAI/OpenAPI-Specification/blob/master/examples/v3.0/petstore.yaml

##### Command line used for generation

java -jar openapi-generator-cli-4.3.1.jar generate -i https://raw.githubusercontent.com/OAI/OpenAPI-Specification/master/examples/v3.0/petstore.yaml -g jaxrs-resteasy -o gen

##### Steps to reproduce

Run the generator and open gen/java/org/openapitools/api/ApiException.java

##### Related issues/PRs

None found

##### Suggest a fix

Adding a basic getter should be enough

### PR DESCRIPTION
**Pull request by Xendar**
Fixes #6830 

Basically a few `ApiException` templates for Java servers were missing a getter for the `code` field.  
Before the fix the generated class was:

```java
package org.openapitools.api;

@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaResteasyServerCodegen", date = "2020-07-01T16:12:55.083+02:00[Europe/Paris]")
public class ApiException extends Exception{
    private int code;
    public ApiException (int code, String msg) {
        super(msg);
        this.code = code;
    }
}
```

With the fix it becomes:

```java
package org.openapitools.api;

@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaResteasyServerCodegen", date = "2020-08-06T22:46:58.073+02:00[Europe/Paris]")
public class ApiException extends Exception {

    private int code;

    public ApiException(int code, String msg) {
        super(msg);
        this.code = code;
    }

    public int getCode() {
        return code;
    }

}
```

### Validation

I compiled the `cli` jar and generated the new class with the same command as in the issue description, then compiled the generated code.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [x] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

Technical committee: @bbdouglas @sreeshas  @jfiala  @lukoyanov @cbornet @jeff9finger @karismann @Zomzog @lwlee2608 @bkabrda

### CODE REVIEW

**Comment by wing328** [comment]
Thanks for the PR but your commit (as shown in the Commits tab) is not linked to your Github account, which means this PR won't count as your contribution in https://github.com/OpenAPITools/openapi-generator/graphs/contributors.

Let me know if you need help fixing it.

Ref: https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-can-i-update-commits-that-are-not-linked-to-my-github-account

File: modules/openapi-generator/src/main/resources/JavaJaxRS/ApiException.mustache [hunk 1]
```diff
@@ -1,10 +1,17 @@
...
+
     private int code;
-    public ApiException (int code, String msg) {
+
+    public ApiException(int code, String msg) {
         super(msg);
         this.code = code;
     }
+
+    public int getCode() {
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/JavaJaxRS/ApiException.mustache:13]: Would be nice if you can add the proper javadoc as well?

Comment (Xendar) [reply modules/openapi-generator/src/main/resources/JavaJaxRS/ApiException.mustache:13]: Sure thing!

**Comment by Xendar** [comment]
> 
> 
> Thanks for the PR but your commit (as shown in the Commits tab) is not linked to your Github account, which means this PR won't count as your contribution in https://github.com/OpenAPITools/openapi-generator/graphs/contributors.
> 
> Let me know if you need help fixing it.
> 
> Ref: https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-can-i-update-commits-that-are-not-linked-to-my-github-account

**Comment by Xendar** [comment]
The Circle CI build failed because of a timeout, but I do not have the rights to relaunch it it seems :(

**Pull request merged**